Lib/unittest/test/test_result.py

```diff
@@ -413,6 +413,47 @@ class tb_frame(object):
         Frame.tb_frame.f_globals['__unittest'] = True
         self.assertTrue(result._is_relevant_tb_level(Frame))
 
+    def testRemoveUnittestTbFrames(self):
+        try:
+            self.fail('too bad')
+        except:
```

Comment:
```suggestion
        except self.failureException:
```
We do not want to swallow KeyboardInterrupt etc.